osf/models/node.py

```diff
@@ -1189,6 +1189,9 @@ def add_contributor(self, contributor, permissions=None, visible=True,
         if contrib_to_add.is_disabled:
             raise ValidationValueError('Deactivated users cannot be added as contributors.')
 
+        if not contrib_to_add.is_registered and not contrib_to_add.unclaimed_records:
+            raise ValidationValueError('Unregistered users cannot be added as contributors without unclaimed records.')
```

Comment:
I also have mixed feelings about this error message. The above "Deactivated" message does get returned by the API when you try and add a deactivated user, so I assume this one will too. The problem is that there should never be unreg users without unclaimed records so it is meaningless for a user of the api user and is not actionable other than "contact support"

Reply:
This isn't really a user facing error this is for developers who wounldn't so they avoid using this internally. Most devs aren't going to check the content of the add_contributor function to ensure they are using it correctly so this is a failsafe.

Reply:
Just in case this does happen again, since this could theoretically get raised might it make sense to change the error message to something more user friendly and generic, with a support link? Then the error would still get raised for devs but a user wouldn't see a confusing message.